### ISSUE TITLE
The cluster config disappeared and re-appeared many times during the day 

### ISSUE THREAD

**Issue by kthommandra** [emissary-ingress/emissary#2974]
**Describe the bug**

Today one of our Ambassador instances cleared all the listeners and clusters resulting in loss of connectivity. This happened multiple times. Upon checking the /ambassador/snapshots/ directory I see some aconf*.yaml files which are very small. Our cluster has more than 200 services. The snapshot.yaml corresponding to the aconf-x.json was fine (had all the cluster info). So I'm guessing this is some Ambassador config generation issue/bug. The small aconf has only the debug "diagnostics" and "internal" information.

**To Reproduce**
The issue happened multiple times today. Never happened before.
![image (1)](https://user-images.githubusercontent.com/2181350/92844882-31d8e580-f39b-11ea-9524-d3a0ce5cfc55.png)

**Expected behavior**
The configuration should not have been removed

**Versions (please complete the following information):**
 - Ambassador: 1.3.1
 - Kubernetes environment : on-prem 1.15.2

**Additional context**
Following problems potentially led to diagd generating a truncated snapshot.yaml 

1) issue with chunked decoding resulting in the load_url_contents terminating abruptly 
2) diagd handling the decode exception and continuing

Another issue:

The incomplete snapshot yaml was able to flow through the snapshot -> aconf -> IR -> econf pipeline to Envoy resulting in the cluster configuration from being removed .. ultimately resulting in "connection refused" to apps.

Strangely enough this issue happened on only 1 Ambassador pod out of many in the same cluster.

**Comment by kthommandra** [comment]
for conversation around it see https://datawire-oss.slack.com/archives/CAULN7S76/p1599781315308700

**Comment by kflynn** [comment]
From the Slack conversation, this error message is in the logs:

```
2020-09-10T14:01:20-07:00 {} 2020-09-10 21:01:20 diagd 1.3.1-205-g814a85761 [P918TAEW] ERROR: couldn't read Kubernetes resources: 'utf-8' codec can't decode byte 0xe2 in position 65535: unexpected end of data
```

That error message is from `_load_url_contents` in `python/ambassador/utils.py`, and as the exception indicates, it is failing to decode a UTF-8 codepoint. The proximate cause here is that `_load_url_contents` tries to decode UTF-8 one 64KiB chunk at a time, and if a UTF-8 codepoint spans the boundary of the 64KiB chunk, it's going to fail. When that happens, higher-level logic will treat it as an empty configuration rather than discarding it entirely. (An empty configuration still makes a certain amount of theoretical sense, but a parse error is not an empty configuration!)

The root cause is currently unknown, since this cascade gets triggered only for non-ASCII codepoints in a `watt` snapshot, and it seems strange that we'd get that. However, we can prevent the cascade happening without being sure of the root cause.

**Comment by kthommandra** [comment]
We just hit this again today. All config except the internal and diagnostics. 

However I did not see the "ERROR: couldn't read Kubernetes resources:" in the Ambassador log file.

There are no ERROR messages in the Ambassador log file. Suddenly there were 404s and then suddenly everything was back to normal.

Upon checking the sizes and contents of the aconf and other conf versions in /ambassador/snapshots I saw that they are very small. By the time I came around to coping these short versions away ... they got rotated.

Now we are running 

$ ambassador --version
Ambassador 1.9.1
Ambassador Scout version 1.9.1
Ambassador Scout semver  1.9.1

envoy  version: 9bf3096803311118a35bc78a39c86b61f3558f5d/1.15.1/Modified/RELEASE/BoringSSL


Any tips on what can we collect from Ambassador/Envoy, next time this happens.
